embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/KafkaDataFormatsTest.java

```diff
@@ -0,0 +1,716 @@
+ * <ul>
+ * Avro (with and without schema registry)
+ * CSV
+ * JSON
+ * Protobuf (with and without schema registry)
+ * </ul>
+ *
+ * This tests both InputFormat and Parser. Parser is deprecated for Kafka Ingestion, and those tests will be removed in the future.
+ */
+public class KafkaDataFormatsTest extends EmbeddedClusterTestBase
+{
+  private static final long CYCLE_PADDING_MS = 100;
+  private static final int EVENTS_PER_SECOND = 6;
+
+
+  private static ObjectMapper jsonMapper;
+  private final EmbeddedBroker broker = new EmbeddedBroker();
+  private final EmbeddedIndexer indexer = new EmbeddedIndexer();
+  private final EmbeddedOverlord overlord = new EmbeddedOverlord();
+  private final EmbeddedHistorical historical = new EmbeddedHistorical();
+  private final EmbeddedCoordinator coordinator = new EmbeddedCoordinator();
+  private KafkaResource kafkaServer;
+  private SchemaRegistryResource schemaRegistry;
+
+  @BeforeAll
+  public static void beforeAll()
+  {
+    jsonMapper = TestHelper.JSON_MAPPER;
+  }
+
+  @Override
+  public EmbeddedDruidCluster createCluster()
+  {
+    final EmbeddedDruidCluster cluster = EmbeddedDruidCluster.withEmbeddedDerbyAndZookeeper().useContainerFriendlyHostname();
+
+    kafkaServer = new KafkaResource()
+    {
+      @Override
+      public void start()
+      {
+        super.start();
+      }
+
+      @Override
+      public void stop()
+      {
+        super.stop();
+      }
+    };
+
+    schemaRegistry = new SchemaRegistryResource()
+    {
+      @Override
+      public void start()
+      {
+        this.kafkaContainer = kafkaServer.getContainer();
+        super.start();
+      }
+
+      @Override
+      public void stop()
+      {
+        super.stop();
+      }
+    };
+
+    indexer.addProperty("druid.segment.handoff.pollDuration", "PT0.1s")
+           .addProperty("druid.worker.capacity", "10");
+    overlord.addProperty("druid.indexer.task.default.context", "{\"useConcurrentLocks\": true}")
+            .addProperty("druid.manager.segments.useIncrementalCache", "ifSynced")
+            .addProperty("druid.manager.segments.pollDuration", "PT0.1s")
+            .addProperty("druid.manager.segments.killUnused.enabled", "true")
+            .addProperty("druid.manager.segments.killUnused.bufferPeriod", "PT0.1s")
+            .addProperty("druid.manager.segments.killUnused.dutyPeriod", "PT1s");
+    coordinator.addProperty("druid.manager.segments.useIncrementalCache", "ifSynced");
+    cluster.addExtension(KafkaIndexTaskModule.class)
+           .addExtension(ProtobufExtensionsModule.class)
+           .addExtension(AvroExtensionsModule.class)
+           .useLatchableEmitter()
+           .addCommonProperty("druid.monitoring.emissionPeriod", "PT0.1s")
+           .addCommonProperty(
+               "druid.monitoring.monitors",
+               "[\"org.apache.druid.java.util.metrics.JvmMonitor\","
+               + "\"org.apache.druid.server.metrics.TaskCountStatsMonitor\"]"
+           )
```

Comment:
We can remove these if monitoring is not needed.

Reply:
The current implementation does wait on a task count metric, but I removed the jvm monitor.